machine/v1beta1/types_azureprovider.go

```diff
@@ -211,6 +218,18 @@ type SecurityProfile struct {
 	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
 }
 
+// AzureSharedGalleryImage defines an image in a Shared Image Gallery to use for VM creation.
+type AzureSharedGalleryImage struct {
```

Comment:
inside this type, is empty string a valid value that is distinct from "no opinion"?

Reply:
empty string would only be valid if all the fields were empty. If AzureSharedGallery images are to be used, all fields are required.